Opcodes/CMakeLists.txt

```diff
@@ -22,20 +22,48 @@ option(BUILD_CPP_OPCODES "Build CPP plugins" ON)
 if(BUILD_CPP_OPCODES)
 message(STATUS "## Building Plugin Opcodes ##")
 ## CPP OPCODE LIBS WITH NO EXTERNAL DEPENDENCIES ##
-make_plugin(doppler doppler.cpp)
+make_plugin(doppler doppler.cpp libcsound_internals)
+if(INIT_STATIC_MODULES)
+    target_compile_defintions(doppler PRIVATE "INIT_STATIC_MODULES")
+endif()
+
 make_plugin(fractalnoise tl/fractalnoise.cpp)
+if(INIT_STATIC_MODULES)
+    target_compile_defintions(fractalnoise PRIVATE "INIT_STATIC_MODULES")
+endif()
 
 if(DIRENT_H) 
-make_plugin(ftsamplebank ftsamplebank.cpp)
+make_plugin(ftsamplebank ftsamplebank.cpp libcsound_internals)
 get_filename_component(DIRENT_INCLUDE_DIR ${DIRENT_H} DIRECTORY)
 target_include_directories(ftsamplebank PRIVATE ${DIRENT_INCLUDE_DIR})
+if(INIT_STATIC_MODULES)
+    target_compile_defintions(ftsamplebank PRIVATE "INIT_STATIC_MODULES")
+endif()
+if(WIN32)
+    target_compile_definitions(ftsamplebank PRIVATE "WIN32")
+endif()
 endif()
 
 make_plugin(lfsr lfsr.cpp)
 make_plugin(bformdec2 bformdec2.cpp)
-make_plugin(mixer mixer.cpp)
-make_plugin(signalflowgraph signalflowgraph.cpp)
-make_plugin(ampmidid ampmidid.cpp)
+if(BIG_ENDIAN)
+    target_compile_definitions(bformdec2 INTERFACE "WORDS_BIGENDIAN")
+endif()
+
+make_plugin(mixer mixer.cpp libcsound_internals)
+if(INIT_STATIC_MODULES)
+    target_compile_defintions(mixer PRIVATE "INIT_STATIC_MODULES")
+endif()
```

Comment:
I don't think this works. I'd prefer that make_plugin would internally make sure of init_static_modules. Rather than manually adding this flag everywhere. But still, this is a big architecture change so I think maybe we should omit that from this PR if this is to get merged anytime soon.

Reply:
Hmm, I'm not sure what you mean by it doesn't work...

> I'd prefer that make_plugin would internally make sure of init_static_modules

But not all plugins need `INIT_STATIC_MODULES`?

> this is a big architecture change so I think maybe we should omit that

Sure, we can just do the first commit if you want

Reply:
Ok looking at it more. I don't think it can be done another way. I'd maybe just group all of these target_compile_definitions in one block. Probably at the end of all these definitions. I thought there might be a more Cmake friendly idiom to create a plugin statically vs dynamically but it would create a lot of changes from what is there now.

Reply:
Ok, done in #1838